# Cleansing Data Youtube Rewind 2015

#### Import Library

In [31]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [32]:
# Memuat value dari file .env
load_dotenv()

mongodb_url = os.getenv('URL_SANDY')
local_url = os.getenv('URL_LOCAL')

In [33]:
database_name = "youtube_rewind_indonesia"
collection_from = "collect_2015"
collection_name = "cleansing_2015"

#### Import dataset dari MongoDB

In [17]:
# Koneksi ke MongoDB
client = MongoClient(local_url)

# Pilih database dan koleksi
db = client[database_name]
collection = db[collection_from]

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [18]:
df

_id                 channelId      videoId  \
0      66515de230ea1caf68a1005c  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
1      66515de230ea1caf68a1005d  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
2      66515de230ea1caf68a1005e  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
3      66515de230ea1caf68a1005f  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
4      66515de230ea1caf68a10060  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
...                         ...                       ...          ...   
28409  66515e2230ea1caf68a16f55  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28410  66515e2230ea1caf68a16f56  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28411  66515e2230ea1caf68a16f57  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28412  66515e2230ea1caf68a16f58  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28413  66515e2230ea1caf68a16f59  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   

                                             textDisplay  \
0                              Masih the best di hati oe   
1                                                  2024🎉   
2      <a href="https://www.youtube.com/watch?v=fNX6G...   
3      Diantara semua YouTube rewind, tahun ini yg pa...   
4      ada yang balik kesini di 2024 ini rewind terba...   
...                                                  ...   
28409                                          Mantep :v   
28410                                             Boom !   
28411                            anjing canggih waaahhhh   
28412                                              firat   
28413                                                  1   

                                            textOriginal  \
0                              Masih the best di hati oe   
1                                                  2024🎉   
2                                  3:34 scene yg gw suka   
3      Diantara semua YouTube rewind, tahun ini yg pa...   
4      ada yang balik kesini di 2024 ini rewind terba...   
...                                                  ...   
28409                                          Mantep :v   
28410                                             Boom !   
28411                            anjing canggih waaahhhh   
28412                                              firat   
28413                                                  1   

            authorDisplayName  \
0         @KaylaAzzahra-fl9sf   
1             @rezapratama391   
2                @acesabo9179   
3                  @WINHERO10   
4            @nadyaamalia1486   
...                       ...   
28409       @GantplayBlogspot   
28410       @ramandahafiz6293   
28411                @e4rtz84   
28412  @pricillaoktaviani2376   
28413       @abayartsgame4609   

                                   authorProfileImageUrl  \
0      https://yt3.ggpht.com/ytc/AIdro_n7417ek-18wzJI...   
1      https://yt3.ggpht.com/ytc/AIdro_kt7csXE65KUFFl...   
2      https://yt3.ggpht.com/ytc/AIdro_lVQpw_BrtYzZYl...   
3      https://yt3.ggpht.com/2S9TD1tTD65kDm2AAWagbPeA...   
4      https://yt3.ggpht.com/ytc/AIdro_koGD1zvhTLlTLo...   
...                                                  ...   
28409  https://yt3.ggpht.com/ytc/AIdro_nyQqIuJWFK6Ie5...   
28410  https://yt3.ggpht.com/ytc/AIdro_nikMBBogL10baS...   
28411  https://yt3.ggpht.com/HWmDyGnG2Kzsde0_Z8ovBbWv...   
28412  https://yt3.ggpht.com/ytc/AIdro_lIX3-HU6zaCl0Z...   
28413  https://yt3.ggpht.com/ytc/AIdro_lwfz4EPeSvrCmj...   

                                    authorChannelUrl  \
0         http://www.youtube.com/@KaylaAzzahra-fl9sf   
1             http://www.youtube.com/@rezapratama391   
2                http://www.youtube.com/@acesabo9179   
3                  http://www.youtube.com/@WINHERO10   
4            http://www.youtube.com/@nadyaamalia1486   
...                                              ...   
28409       http://www.youtube.com/@GantplayBlogspot   
28410       http://www.youtube.com/@ramandahafiz6293   
28411                http://www.youtube.com/@e4rtz84   
28412  http://www.y

#### Menghapus data text yang duplikat

In [23]:
df = df.drop_duplicates(subset=['textOriginal'])

df.shape

(26486, 14)

#### Menghapus data author yang duplikat

In [24]:
df = df.drop_duplicates(subset=['authorChannelUrl'])

df.shape

(24317, 14)

#### Menghilangkan atribut yang tidak diperlukan

Karena hanya menganalisis komentar, maka atribut yang disisakan hanya _id dan textOriginal

In [25]:
df = df[['_id', 'textOriginal']]
df

_id  \
0      66515de230ea1caf68a1005c   
1      66515de230ea1caf68a1005d   
2      66515de230ea1caf68a1005e   
3      66515de230ea1caf68a1005f   
4      66515de230ea1caf68a10060   
...                         ...   
28401  66515e2230ea1caf68a16f4d   
28402  66515e2230ea1caf68a16f4e   
28408  66515e2230ea1caf68a16f54   
28409  66515e2230ea1caf68a16f55   
28411  66515e2230ea1caf68a16f57   

                                            textOriginal  
0                              Masih the best di hati oe  
1                                                  2024🎉  
2                                  3:34 scene yg gw suka  
3      Diantara semua YouTube rewind, tahun ini yg pa...  
4      ada yang balik kesini di 2024 ini rewind terba...  
...                                                  ...  
28401                                              woeee  
28402                                                  1  
28408                                        =))) kerenn  
28409                                          Mantep :v  
28411                            anjing canggih waaahhhh  

[24317 rows x 2 columns]

#### Menghapus nilai yang kosong

In [26]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id             0
textOriginal    0
dtype: int64

Dari yang awalnya berjumlah 28414 row berkurang menjadi 24317 row

#### Mengimpan Clean Comments ke MongoDB Local

In [27]:
# Koneksi ke MongoDB
client = MongoClient(local_url)
db = client[database_name]
collection = db[collection_name]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Local.")

Data berhasil diimpor ke MongoDB Local.


#### Migrasi dari MongoDB Local ke MongoDB Atlas

In [35]:
# Koneksi ke MongoDB lokal
local_client = MongoClient(local_url)
local_db = local_client[database_name]
local_collection = local_db[collection_name]

# Koneksi ke MongoDB Atlas
atlas_client = MongoClient(mongodb_url)
atlas_db = atlas_client[database_name]
atlas_collection = atlas_db[collection_name]

# Ambil semua dokumen dari koleksi MongoDB lokal
documents = list(local_collection.find())

# Masukkan dokumen ke MongoDB Atlas
if documents:
    atlas_collection.insert_many(documents)
    print(f'{len(documents)} dokumen berhasil dipindahkan dari MongoDB lokal ke MongoDB Atlas.')
else:
    print('Tidak ada dokumen yang ditemukan di MongoDB lokal.')

# Tutup koneksi
local_client.close()
atlas_client.close()

OperationFailure: you are over your space quota, using 515 MB of 512 MB, full error: {'ok': 0, 'errmsg': 'you are over your space quota, using 515 MB of 512 MB', 'code': 8000, 'codeName': 'AtlasError'}